engine/gamesys/src/gamesys/scripts/script_resource.cpp

```diff
@@ -2088,6 +2129,84 @@ static const luaL_reg Module_methods[] =
  * @variable
  */
 
+/*# RGB_PVRTC_2BPPV1 type texture format
+ *
+ * @name resource.TEXTURE_FORMAT_RGB_PVRTC_2BPPV1
+ * @variable
+ */
+
+/*# RGB_PVRTC_4BPPV1 type texture format
+ *
+ * @name resource.TEXTURE_FORMAT_RGB_PVRTC_4BPPV1
+ * @variable
+ */
+
+/*# RGBA_PVRTC_2BPPV1 type texture format
+ *
+ * @name resource.TEXTURE_FORMAT_RGBA_PVRTC_2BPPV1
+ * @variable
+ */
+
+/*# RGBA_PVRTC_4BPPV1 type texture format
+ *
+ * @name resource.TEXTURE_FORMAT_RGBA_PVRTC_4BPPV1
+ * @variable
+ */
+
+/*# RGB_ETC1 type texture format
+ *
+ * @name resource.TEXTURE_FORMAT_RGB_ETC1
+ * @variable
+ */
+
+/*# RGBA_ETC2 type texture format
+ *
+ * @name resource.TEXTURE_FORMAT_RGBA_ETC2
+ * @variable
+ */
+
+/*# RGBA_ASTC_4x4 type texture format
+ *
+ * @name resource.TEXTURE_FORMAT_RGBA_ASTC_4x4
+ * @variable
+ */
+
+/*# RGB_BC1 type texture format
+ *
+ * @name resource.TEXTURE_FORMAT_RGB_BC1
+ * @variable
+ */
+
+/*# RGBA_BC3 type texture format
+ *
+ * @name resource.TEXTURE_FORMAT_RGBA_BC3
+ * @variable
+ */
+
+/*# R_BC4 type texture format
+ *
+ * @name resource.TEXTURE_FORMAT_R_BC4
+ * @variable
+ */
+
+/*# RG_BC5 type texture format
+ *
+ * @name resource.TEXTURE_FORMAT_RG_BC5
+ * @variable
+ */
+
+/*# RGBA_BC7 type texture format
+ *
+ * @name resource.TEXTURE_FORMAT_RGBA_BC7
+ * @variable
+ */
+
+/*# BASIS_UASTC compression type
+ *
+ * @name resource.COMPRESSION_TYPE_BASIS_UASTC
+ * @variable
+ */
```

Comment:
Does the default texture compression need to be listed?

Reply:
Not sure what you mean, do you mean that we should mark which one of these that is the default value in the docs?